botocore/docs/method.py

```diff
@@ -207,6 +208,17 @@ def document_model_driven_method(section, method_name, operation_model,
                 request_params_section, operation_model.input_shape,
                 include=include_input, exclude=exclude_input)
 
+
+    # Find idempotent params and append documentation
+    if 'request-params' in section.available_sections:
+        param_section = section.get_section('request-params')
+        for param_name in param_section.available_sections:
```

Comment:
I would prefer we move this in ``params.py`` and ``shape.py``.  It seems weird to be adding parameter specific logic not in one of these files, especially with the use of the context (it seems unneeded right now). You can probably follow the same logic as streaming input shape.